sigil.lic

```diff
@@ -0,0 +1,153 @@
+    when 'Having recently been searched'
+      exit
+    when *@primary_sigils
+      return true if line.include?(@target)
+      find_sigil?
+    when *@secondary_sigils
+      return true if line.include?(@target)
+      return false
+    when 'Roundtime'
+      find_sigil?
+    end
+  end
+
+  def scribe_sigil
+    Flags.reset('moresigils')
+    bput('get my burin', 'You get', 'You are already')
+    bput('get my blank scrolls', 'You get', 'You are already','You pick up')
+    bput('scribe sigil', 'You need', 'You carefully scribe')
+    waitrt?
+    bput('stow burin', 'You put')
+    bput("get #{@sigil_book}", 'You get')
+
+    case bput("put sigil in my #{@sigil_book}", 'You add', 'You rearrange', 'completely full')
+    when 'completely full'
+      bput('stow sigil', 'You put')
+    end
+    bput('stow blank scrolls', 'You put')
+    bput("stow my #{@sigil_book}", 'You put')
+    @equipment_manager.empty_hands
+    if Flags['moresigils']
+      scribe_sigil
+    end
+  end
+  
+  def get_potential_actions
+    waitrt?
+    potential_actions = []
+    snapshot = reget(50)
+    snapshot = snapshot.reverse
+    
+    snapshot.each do | line |
+      if line.include?('sigil precision')
+        potential_actions << line
+      end
+      if line.include?("You have perceived a")
+        break
+      end
+    end
+    return potential_actions
+  end
+ 
+  def refine_sigil
+    bput('perc sigil improve','You have perceived')
+    @current_precision = Flags['precision'][1].to_i
+    
+    while @current_precision < @target_precision
+      potential_actions = get_potential_actions
+      potential_actions.each do | line |
+        if line =~ /...a \w+, (\w+) \w+ ([A-Z]+)/
+          waitrt?
+          
+          return unless can_continue?
+          
+          stat = Regexp.last_match(1)
+          action = Regexp.last_match(2)
+          
+          next if Flags[stat][2].length() <=3
+          # echo "Performing action that will hurt #{stat}.  #{stat} is currently at #{Flags[stat][2].length()}"
+          
+          Flags.reset('clarity')
+          Flags.reset('precision')
+          Flags.reset('danger')
+          Flags.reset('sanity')
+          Flags.reset('resolve')
+          Flags.reset('focus')
+          
+          if action=="ACTION"
+            fput("perc sigil #{action}")
+            waitrt?
+          end
+          case bput("perc sigil #{action}","Your mind survives a chaotic break","You successfully resolve","The strain of perceiving sigils bears down","You are unaware of any sigil's capable of that method","Your focus successfully weathers","Your resolve collapses","Chills creep down your spine","A sudden sneeze and sinus pain assault your senses","You are unaware of any sigils in the area","You prepare yourself for continued exertion","the sigil is nowhere to be found","No traces of the sigil remain","You are unaware of any sigils in the area","The world dissolves to chaos")
```

Comment:
Added two more catches:
`"You have perceived a flawless", "You have perceived an exquisite"`.

```suggestion
          case bput("perc sigil #{action}",
                    "Your mind survives a chaotic break",
                    "You successfully resolve",
                    "The strain of perceiving sigils bears down",
                    "You are unaware of any sigil's capable of that method",
                    "Your focus successfully weathers",
                    "Your resolve collapses",
                    "Chills creep down your spine",
                    "A sudden sneeze and sinus pain assault your senses",
                    "You are unaware of any sigils in the area",
                    "You prepare yourself for continued exertion",
                    "the sigil is nowhere to be found",
                    "No traces of the sigil remain",
                    "You are unaware of any sigils in the area",
                    "The world dissolves to chaos",
                    "You have perceived a flawless", 
                     "You have perceived an exquisite")
```